src/pybind/mgr/dashboard/controllers/host.py

```diff
@@ -151,9 +155,25 @@ def merge_hosts_by_hostname(ceph_hosts, orch_hosts):
             }, orch_hosts_map[hostname]) for hostname in orch_hosts_map
     ]
     hosts.extend(orch_hosts_only)
+    for host in hosts:
+        host['service_instances'] = json.loads(
+            populate_service_instances(host['hostname'], host['services']))
     return hosts
 
 
+def populate_service_instances(hostname, services):
+    orch = OrchClient.instance()
+    if orch.available():
+        return json.dumps(Counter([
+            daemon['daemon_type'] for daemon in [
+                d.to_dict() for d in orch.services.list_daemons(
+                    hostname=hostname)]]))
+
+    return json.dumps(Counter([
+        daemon['type'] for daemon in services
+    ]))
```

Comment:
I'm not following here: why are we JSON-serializing & deserializing? Internal function-passing should rather use Python objects. Serialization should only be used for networking and interfacing with external components...

Reply:
yeah, my bad here. but updated.